### ISSUE TITLE
training code

### ISSUE THREAD

**Issue by borre47** [jannctu/TIN#7]
Could you share with us the training code and the loss function?
